src/base/http/server.cpp

```diff
@@ -108,9 +109,13 @@ void Server::incomingConnection(const qintptr socketDescriptor)
         static_cast<QSslSocket *>(serverSocket)->startServerEncryption();
     }
 
-    auto *c = new Connection(serverSocket, m_requestHandler, this);
-    m_connections.insert(c);
-    connect(serverSocket, &QAbstractSocket::disconnected, this, [c, this]() { removeConnection(c); });
+    Connection *socket = new Connection(serverSocket, this);
```

Comment:
you should keep the `auto *`
and the variable name is imprecise: it is not a socket, but a connection.